modules/andes-core/client/src/main/java/org/wso2/andes/client/XAConnectionImpl.java

```diff
@@ -124,6 +145,18 @@ public synchronized void close() throws JMSException {
                 super.close();
             } else {
                 LOGGER.error("XAConnection.close() was called before committing or rolling back");
+                connectionCloseFuture = scheduledExecutor.schedule(new Runnable() {
+                    @Override
+                    public void run() {
+                        try {
+                            LOGGER.error("Closing XAConnection after waiting " + CONNECTION_CLOSE_TIMEOUT
+                                                 + " seconds for a commit or rollback");
+                            closeIfNoActiveSessions();
+                        } catch (JMSException e) {
+                            LOGGER.error("Error occurred while closing the XAConnection after close timeout");
+                        }
+                    }
+                }, CONNECTION_CLOSE_TIMEOUT, TimeUnit.SECONDS);
```

Comment:
I count this as 1 vote for a config. :smile:  I do not think we need to configure this timeout in Millisecond's level since this is not a performance tuning parameter rather a fail safe parameter.